projects/plugins/jetpack/extensions/blocks/revue/revue.php

```diff
@@ -31,223 +31,28 @@ function register_block() {
-		'emailPlaceholder'     => __( 'Enter your email address', 'jetpack' ),
-		'firstNameLabel'       => __( 'First name', 'jetpack' ),
-		'firstNamePlaceholder' => __( 'Enter your first name', 'jetpack' ),
-		'firstNameShow'        => true,
-		'lastNameLabel'        => __( 'Last name', 'jetpack' ),
-		'lastNamePlaceholder'  => __( 'Enter your last name', 'jetpack' ),
-		'lastNameShow'         => true,
-	);
-
-	if ( array_key_exists( $attribute, $default_attributes ) ) {
-		return $default_attributes[ $attribute ];
-	}
-}
-
-/**
- * DEPRECATED V1
- */
-
-/**
- * Create the Revue subscribe button.
- *
  * @param array $attributes Array containing the Revue block attributes.
  *
  * @return string
  */
-function get_deprecated_v1_revue_button( $attributes ) {
-	$classes = array( 'wp-block-button__link' );
-	$styles  = array();
-
-	$text                        = get_revue_attribute( 'text', $attributes );
-	$has_class_name              = array_key_exists( 'className', $attributes );
-	$has_named_text_color        = array_key_exists( 'textColor', $attributes );
-	$has_custom_text_color       = array_key_exists( 'customTextColor', $attributes );
-	$has_named_background_color  = array_key_exists( 'backgroundColor', $attributes );
-	$has_custom_background_color = array_key_exists( 'customBackgroundColor', $attributes );
-	$has_named_gradient          = array_key_exists( 'gradient', $attributes );
-	$has_custom_gradient         = array_key_exists( 'customGradient', $attributes );
-	$has_border_radius           = array_key_exists( 'borderRadius', $attributes );
-
-	if ( $has_class_name ) {
-		$classes[] = $attributes['className'];
-	}
-
-	if ( $has_named_text_color || $has_custom_text_color ) {
-		$classes[] = 'has-text-color';
-	}
-	if ( $has_named_text_color ) {
-		$classes[] = sprintf( 'has-%s-color', $attributes['textColor'] );
-	} elseif ( $has_custom_text_color ) {
-		$styles[] = sprintf( 'color: %s;', $attributes['customTextColor'] );
+function render_block( $attributes ) {
+	if ( current_user_can( 'manage_options' ) ) {
+		$wpcom_newsletter_url = esc_url( 'https://wordpress.com/go/digital-marketing/migrate-from-revue-newsletter/' );
+		$message              = sprintf(
+			'<br/>'
+			/* Translators: %s contains the words 'shutting down', which links to a Revue help article about the feature being shut down. */
+			. esc_html__( 'Revue is shutting down. The Revue signup form will no longer be displayed to your visitors and as such this block should be removed. %1$s You can migrate from Revue to the WordPress.com Newsletter - %2$s.', 'jetpack' ),
+			'<br/> <br/>',
+			sprintf(
+				'<a href="%1$s" target="_blank">%2$s</a>',
+				$wpcom_newsletter_url,
+				esc_html__( 'find out more here', 'jetpack' )
+			)
+		);
+		return sprintf(
+			'<div class="jetpack-block__notice info %1$s" style="border-left:5px solid #dba617;padding:1em;background-color:#f8f9f9;">%2$s</div>',
+			esc_attr( Blocks::classes( FEATURE_NAME, $attributes ) ),
+			$message
+		);
```

Comment:
The mix of `sprintf`, translatable strings, and html tags seems a bit too complex for me. I think it could be made more readable. We could also take the opportunity to use the existing `Jetpack_Gutenberg::notice()` for the output to avoid duplication. 

Here is a diff you can apply to show you what I have in mind. This has a few advantages:

- The translatable strings now match the ones in the editor; that means the translators will only have to translate once, instead of translating strings that are *almost* the same.
- We use `Jetpack_Gutenberg::notice()` as I mentioned above.
- This is subjective, but I think it's more readable.

```diff
diff --git a/projects/plugins/jetpack/class.jetpack-gutenberg.php b/projects/plugins/jetpack/class.jetpack-gutenberg.php
index 15001f1b15..b942c65b89 100644
--- a/projects/plugins/jetpack/class.jetpack-gutenberg.php
+++ b/projects/plugins/jetpack/class.jetpack-gutenberg.php
@@ -1084,6 +1084,11 @@ class Jetpack_Gutenberg {
 				array(
 					'br' => array(),
 					'p'  => array(),
+					'a'  => array(
+						'href'   => array(),
+						'target' => array(),
+						'rel'    => array(),
+					),
 				)
 			),
 			esc_attr( $classes ),
diff --git a/projects/plugins/jetpack/extensions/blocks/revue/revue.php b/projects/plugins/jetpack/extensions/blocks/revue/revue.php
index 585c197e68..2746c489c1 100644
--- a/projects/plugins/jetpack/extensions/blocks/revue/revue.php
+++ b/projects/plugins/jetpack/extensions/blocks/revue/revue.php
@@ -10,6 +10,7 @@
 namespace Automattic\Jetpack\Extensions\Revue;
 
 use Automattic\Jetpack\Blocks;
+use Jetpack_Gutenberg;
 
 const FEATURE_NAME = 'revue';
 const BLOCK_NAME   = 'jetpack/' . FEATURE_NAME;
@@ -35,24 +36,20 @@ add_action( 'init', __NAMESPACE__ . '\register_block' );
  *
  * @return string
  */
-function render_block( $attributes ) {
+function render_block( $attributes ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
 	if ( current_user_can( 'manage_options' ) ) {
-		$wpcom_newsletter_url = esc_url( 'https://wordpress.com/go/digital-marketing/migrate-from-revue-newsletter/' );
-		$message              = sprintf(
-			'<br/>'
-			/* Translators: %s contains the words 'shutting down', which links to a Revue help article about the feature being shut down. */
-			. esc_html__( 'Revue is shutting down. The Revue signup form will no longer be displayed to your visitors and as such this block should be removed. %1$s You can migrate from Revue to the WordPress.com Newsletter - %2$s.', 'jetpack' ),
-			'<br/> <br/>',
-			sprintf(
-				'<a href="%1$s" target="_blank">%2$s</a>',
-				$wpcom_newsletter_url,
-				esc_html__( 'find out more here', 'jetpack' )
-			)
+		$message  = esc_html__( 'Revue is shutting down. The Revue signup form will no longer be displayed to your visitors and as such this block should be removed.', 'jetpack' );
+		$message .= '<br/><br/>';
+		$message .= sprintf(
+			' <a href="%1$s" target="_blank" rel="noopener noreferrer">%2$s</a>',
+			esc_url( 'https://wordpress.com/go/digital-marketing/migrate-from-revue-newsletter/' ),
+			esc_html__( 'You can migrate from Revue to the WordPress.com Newsletter - find out more here.', 'jetpack' )
 		);
-		return sprintf(
-			'<div class="jetpack-block__notice info %1$s" style="border-left:5px solid #dba617;padding:1em;background-color:#f8f9f9;">%2$s</div>',
-			esc_attr( Blocks::classes( FEATURE_NAME, $attributes ) ),
-			$message
+
+		return Jetpack_Gutenberg::notice(
+			$message,
+			'warning',
+			Blocks::classes( FEATURE_NAME, $attributes )
 		);
 	}
 }
```

Reply:
Ha, thanks @jeherve !  I'd thought I'd work out how to add links to Jetpack_Gutenberg::notice for future use separately, but no need now with this! I'll take a look and follow-up.

Reply:
Looks good now, thanks for looking at this.